frontend/src/utils/ews-utils.ts

```diff
@@ -57,16 +58,19 @@ type EWSTriggerLevels = {
 export const createEWSDatesArray = (testEndDate?: number): number[] => {
   const datesArray = [];
 
-  const endDate =
-    testEndDate || moment(moment.utc().format('YYYY-MM-DD')).valueOf();
+  const now = new Date();
 
-  const tempDate = moment.utc('2021-01-01');
+  const endDate = testEndDate || now.setUTCHours(0, 0, 0, 0);
 
-  while (tempDate.valueOf() <= endDate) {
+  let tempDate = new Date('2021-01-01');
+
+  while (tempDate.getTime() <= endDate) {
+    const clone = new Date(tempDate.getTime());
     // eslint-disable-next-line fp/no-mutating-methods
-    datesArray.push(tempDate.clone().set({ hour: 12, minute: 0 }).valueOf());
+    datesArray.push(clone.setUTCHours(12, 0, 0, 0));
 
-    tempDate.add(1, 'days');
+    // eslint-disable-next-line fp/no-mutation
+    tempDate = new Date(tempDate.getTime() + oneDayInMs);
```

Comment:
It's a bit of a weird corner case, but I wonder how much of an impact the [leap second](https://en.wikipedia.org/wiki/Leap_second) has on our date management. It looks like there's only been 5 in the past 20 years, so likely nothing to worry about :crossed_fingers: 

That being said, I would be tempted to modify this `while` loop to add 1 day at a time with a counter, and do `datesArray.push(tempDate.setUTCDate(counter))` so you can avoid creating all these temp date clones.

Reply:
we are setting hours to 12, so leap second can't change the date, i believe